templates/aviation-checklist/pdf_aviation_checklist.go

```diff
@@ -0,0 +1,249 @@
+		drawHeader("templates/header.tpl", block, checks.Version, checks.ReleaseDate, args.PageNum, args.TotalPages)
+	})
+	c.DrawFooter(func(block *creator.Block, args creator.FooterFunctionArgs) {
+		drawHeader("templates/footer.tpl", block, checks.Version, checks.ReleaseDate, args.PageNum, args.TotalPages)
+	})
+
+	// Write output file.
+	if err := c.WriteToFile("unipdf-aviation-checklist.pdf"); err != nil {
+		log.Fatal(err)
+	}
+}
+
+// readTemplate reads the template at the specified file path and returns it
+// as an io.Reader.
+func readTemplate(tplFile string) (io.Reader, error) {
+	file, err := os.Open(tplFile)
+	if err != nil {
+		return nil, err
+	}
+	defer file.Close()
+
+	buf := bytes.NewBuffer(nil)
+	if _, err = io.Copy(buf, file); err != nil {
+		return nil, err
+	}
+
+	return buf, nil
+}
+
+// readChecklistData reads the checklist data from a specified JSON file.
+func readChecklistData(jsonFile string) (*Checklist, error) {
+	file, err := os.Open(jsonFile)
+	if err != nil {
+		return nil, err
+	}
+	defer file.Close()
+
+	check := &Checklist{}
+	if err := json.NewDecoder(file).Decode(check); err != nil {
+		return nil, err
+	}
+
+	return check, nil
+}
+
+// Checklist represent checklist data.
+type Checklist struct {
+	Version     string    `json:"version"`
+	ReleaseDate string    `json:"release"`
+	Checks      []*Checks `json:"checklist"`
+}
+
+// Checks represent each group of checklist.
+type Checks struct {
+	Title string       `json:"title"`
+	Items []*CheckItem `json:"items"`
+}
+
+// CheckItem represents each item in checklist groups.
+type CheckItem struct {
+	Action *string `json:"action"`
+	Check  *string `json:"check,omitempty"`
+	Value  *string `json:"value,omitempty"`
+}
+
+// DisplayText returns combination text of checklist label, separator, and value.
+func (c *CheckItem) DisplayText() string {
+	if c.Value == nil {
+		return *c.Check
+	}
+
+	// Max line width.
+	maxWidth := 258.0
+
+	stdFont := model.StdFontName(model.HelveticaName)
+	font, err := model.NewStandard14Font(stdFont)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	textWidth := 0.0
+	for _, r := range *c.Check + *c.Value {
+		textWidth += getRuneWidth(font, r)
+	}
+
+	if textWidth < maxWidth {
+		availWidth := maxWidth - textWidth
+		separator := '.'
+		sepWidth := getRuneWidth(font, separator)
+		sepCount := int(availWidth / sepWidth)
+
+		// Space to narrow for dot separator, use space
+		if sepCount <= 5 {
+			separator = ' '
+			sepWidth := getRuneWidth(font, separator)
+			sepCount = int(availWidth / sepWidth)
+		}
+		sepText := strings.Repeat(string(separator), sepCount)
+
+		return *c.Check + sepText + *c.Value
+	}
+
+	return *c.Check + "  " + *c.Value
+}
+
+// getRuneWidth calculates rune width based on provided font.
+func getRuneWidth(font *model.PdfFont, r rune) float64 {
+	metrics, bool := font.GetRuneMetrics(r)
+	if !bool {
+		log.Fatal("failed to get width")
+	}
+	return metrics.Wx / 100
```

Comment:
What does `100` represent? Should it be `1000.0`?

Reply:
Yes you are correct, normally this should be 1000.0. It use 100 as a shortcut to include font size in the width calculation. The font size used in the checklist text is 10.

I've rewrite the code to be more clear on this calculation.